### DISCUSSION TITLE
Unsuspected seeming large variation between pass #0 duration and that of (some?) subsequent passes.

### DISCUSSION THREAD

**Discussion by ghchris2021** [memtest86plus/memtest86plus#331] [Q&A] [answered]
Re: Unsuspected seeming large variation between pass #0 duration and that of (some?) subsequent passes. 

memtest86+ v6.10 listed on screen (included in manjaro installer hybrid usb/iso image)

This may be entirely expected / documented / normal / harmless but I hadn't been aware of this perceived characteristic and it isn't something I would have suspected a-priori.  It's even possible my analysis / interpretation is somehow very faulty.  Anyway I apologize if this is irrelevant.

 I'm just noting itfo in case it is actually unexpected and possibly problematic.

I have noticed per. the below (ongoing) test information that on a given system:

Pass #0 took ~02:14:20 to complete (all tests enabled, SMP 24T PAR).

Pass #1 only reported 35% complete after ~02:14:xx time elapsed since pass #1 ~started;
comparitively pass #0 was reported done after this much elapsed in-pass time.

Presently 14:08:19 has elapsed since starting Pass #0 and it indicates Pass: 3 in progress @ 08% complete (Passes #0, #1, #2 fully complete).

So it having completed Pass #0 in 2h:14m means that (Pass #1 + Pass #2 combined) must have taken ~(14:08:xx-02:14:xx) = 11:54:xx or so estimated (11:54:xx / 2) = 05:57:xx to complete if they each required the ~same amount of time and the indicated "Time" report is roughly correct.

The system does not (AFAIK) have any particularly major power management / sleep / suspend or thermal conditions that I am aware of in effect
such that I would expect the memory throughput or the CPU throughput to change significantly over the operational time.
The CPU temperature is ~constant at ~36-38C every time I've looked, and the CPU clock frequency ~ constant AFAIK. 

So it looks to me like in this case Pass #0 anomalously may have taken perhaps less than half of the time to run than
subsequent passes were estimated to have taken, or at least there can possibly be some other such significant variance in pass execution time.

I would have assumed that even though there is mention of "random" in various test pattern descriptions that the overall test
pass execution work-load scope and duration would be basically equal and deterministic; perhaps (due to reused / reset PRNG seeds or whatever) even
the exact test patterns could possibly be literally numerically identical from pass to pass though even if not perhaps the overall count of memory
reads / writes to a given address range would be ~constant even if the addresses / data patterns might vary.

20230809-0839 Pass #0 started

Started memtest86+ v6.10
12C / SMP 24T (PAR) 1-socket consumer ATX motherboard testing
All tests enabled (default configuration).
3 populated slots of memory 32GBy DDR4 non-ECC each

20230809-0935 Pass #0 listed as 39% complete after Time: 0:56:39

20230809-1050 Pass #0 listed as 99% complete (Time: 02:10:55)

20230809-1053 Pass #0 completed (Time: 02:14:20 Pass: 1 now)

20230809-1312 Pass #1 still running (Pass: 35%; Time: 04:33:25 Pass: 1 still)

20230809-2247 Pass #3 now running (Pass: 08%; Time: 14:08:19)

**Comment by martinwhitaker** [comment, marked as answer]
This is the expected behaviour. On the first pass, the number of iterations within each test is reduced. This allows a relatively quick run through all the tests before commencing the more thorough testing.
